src/bpftrace.cpp

```diff
@@ -573,66 +587,52 @@ void perf_event_printer(void *cb_cookie, void *data, int size __attribute__((unu
     }
 
     auto id = printf_id - asyncactionint(AsyncAction::syscall);
-    auto fmt = std::get<0>(bpftrace->system_args_[id]).c_str();
+    auto fmt = std::get<0>(bpftrace->system_args_[id]);
     auto args = std::get<1>(bpftrace->system_args_[id]);
     auto arg_values = bpftrace->get_arg_values(args, arg_data);
 
-    const int BUFSIZE = 512;
-    char buffer[BUFSIZE];
-    int sz = format(buffer, BUFSIZE, fmt, arg_values);
-    // Return value is required size EXCLUDING null byte
-    if (sz >= BUFSIZE)
+    std::string retstr = format(fmt, arg_values);
+    auto sz = retstr.length();
+    if (sz >= FMT_BUF_SZ)
     {
-      std::cerr << "syscall() command to long (" << sz << " bytes): ";
-      std::cerr << buffer << std::endl;
+      std::cerr << "syscall() command too long (" << sz << " bytes): ";
+      std::cerr << retstr << std::endl;
       return;
     }
-    bpftrace->out_->message(MessageType::syscall, exec_system(buffer), false);
+    bpftrace->out_->message(MessageType::syscall,
+                            exec_system(retstr.c_str()),
+                            false);
     return;
   }
   else if ( printf_id >= asyncactionint(AsyncAction::cat))
   {
     auto id = printf_id - asyncactionint(AsyncAction::cat);
-    auto fmt = std::get<0>(bpftrace->cat_args_[id]).c_str();
+    auto fmt = std::get<0>(bpftrace->cat_args_[id]);
     auto args = std::get<1>(bpftrace->cat_args_[id]);
     auto arg_values = bpftrace->get_arg_values(args, arg_data);
 
-    const int BUFSIZE = 512;
-    char buffer[BUFSIZE];
-    int sz = format(buffer, BUFSIZE, fmt, arg_values);
-    // Return value is required size EXCLUDING null byte
-    if (sz >= BUFSIZE)
+    std::string retstr = format(fmt, arg_values);
+    auto sz = retstr.length();
+    if (sz >= FMT_BUF_SZ)
     {
       std::cerr << "cat() command to long (" << sz << " bytes): ";
-      std::cerr << buffer << std::endl;
+      std::cerr << retstr << std::endl;
       return;
     }
     std::stringstream buf;
-    cat_file(buffer, bpftrace->cat_bytes_max_, buf);
+    cat_file(retstr.c_str(), bpftrace->cat_bytes_max_, buf);
```

Comment:
I think we can drop the FMT_BUF_SZ check. I don't believe it does anything useful anymore.

So instead it might look like:
```
std:stringstream buf;
cat_file(format(fmt, arg_vaules).c_str(), bpftrace->cat_bytes_max_, buf);
```